workers/event_catcher/event_catcher.rb

```diff
@@ -0,0 +1,210 @@
+
+  def connect
+    vim_opts = {
+      :ns       => 'urn:vim25',
+      :host     => hostname,
+      :ssl      => true,
+      :insecure => true,
+      :path     => '/sdk',
+      :port     => 443,
+      :rev      => '6.5',
+    }
+
+    require 'rbvmomi'
+    RbVmomi::VIM.new(vim_opts).tap do |vim|
+      vim.rev = vim.serviceContent.about.apiVersion
+      vim.serviceContent.sessionManager.Login(:userName => username, :password => password)
+    end
+  end
+
+  def create_event_history_collector(vim, page_size)
+    filter = RbVmomi::VIM.EventFilterSpec()
+
+    event_manager = vim.serviceContent.eventManager
+    event_manager.CreateCollectorForEvents(:filter => filter).tap do |c|
+      c.SetCollectorPageSize(:maxCount => page_size)
+    end
+  end
+
+  def create_property_filter(vim, event_history_collector)
+    vim.propertyCollector.CreateFilter(
+      :spec => RbVmomi::VIM.PropertyFilterSpec(
+        :objectSet    => [
+          RbVmomi::VIM.ObjectSpec(
+            :obj => event_history_collector
+          )
+        ],
+        :propSet      => [
+          RbVmomi::VIM.PropertySpec(
+            :type    => event_history_collector.class.wsdl_name,
+            :all     => false,
+            :pathSet => ["latestPage"]
+          )
+        ]
+      ),
+      :partialUpdates => true
+    )
+  end
+
+  def wait_for_updates(vim)
+    version = nil
+    options = RbVmomi::VIM.WaitOptions(:maxWaitSeconds => 60)
+
+    loop do
+      update_set = vim.propertyCollector.WaitForUpdatesEx(:version => version, :options => options)
+      heartbeat
+      next if update_set.nil?
+
+      version = update_set.version
+
+      Array(update_set.filterSet).each do |property_filter_update|
+        Array(property_filter_update.objectSet).each do |object_update|
+          next unless object_update.kind == "modify"
+
+          Array(object_update.changeSet).each do |property_change|
+            yield property_change
+          end
+        end
+      end
+    end
+  end
+
+  def parse_event(event)
+    event_type = event.class.wsdl_name
+
+    result = {
+      :ems_id     => ems_id,
+      :event_type => event_type,
+      :chain_id   => event.chainId,
+      :is_task    => event_type == "TaskEvent",
+      :source     => "VC",
+      :message    => event.fullFormattedMessage,
+      :timestamp  => event.createdTime,
+      :full_data  => event.props
+    }
+
+    result
+  end
+
+  def publish_events(events)
+    events.each do |event|
+      messaging_client.publish_topic(
+        :service => "manageiq.ems-events",
+        :sender  => ems_id,
+        :event   => event[:event_type],
+        :payload => event
+      )
+    end
+  end
+
+  def messaging_client
+    @messaging_client ||= begin
+      ManageIQ::Messaging::Client.open(
+        :host       => messaging_host,
+        :port       => messaging_port,
+        :protocol   => :Kafka,
+        :encoding   => "json",
+        :client_ref => "vmware-event-catcher-#{ems_id}"
```

Comment:
should this be something more detailed like `Vmware::InfraManager::EventCatcher.#{ems_id}` or whatever the kafka syntax allows?